src/audio/output.rs

```diff
@@ -0,0 +1,35 @@
+//! Provides structures and functions used to get audio outputs.
+
+// We have to use types from this to provide an output iterator type.
+extern crate cpal;
```

Comment:
I was reluctant to add a dependency on this crate at all, but it was necessary to write the OutputIterator type, as soon as `impl Trait` is stabilized I'd like to remove this dependency, so I chose to limit the scope of it.  Do you feel my decision making process here was sound?